src/lib.rs

```diff
@@ -0,0 +1,742 @@
+pub mod errors;
+pub mod file_watcher;
+pub mod systeminfo_watcher;
+
+use crate::errors::TelemetryError;
+
+use base64::{engine::general_purpose::STANDARD, Engine};
+use chrono::Utc;
+use dirs::home_dir;
+use nix::{
+    errno::Errno,
+    fcntl::{Flock, FlockArg},
+    sys::stat,
+    unistd::{chdir, close, dup2, fork, setsid, sysconf, SysconfVar},
+};
+use regex::Regex;
+use std::{
+    env::{var, var_os},
+    fs::{create_dir_all, File, OpenOptions},
+    io::{stderr, stdout, Write},
+    os::fd::AsRawFd,
+    path::{Path, PathBuf},
+    process::exit,
+    sync::LazyLock,
+};
+use sysinfo::System;
+use tracing::{
+    span::{Entered, Id, Record},
+    Event, Span, Subscriber,
+};
+use tracing_appender::non_blocking::{NonBlocking, WorkerGuard};
+use tracing_subscriber::layer::Context;
+use tracing_subscriber::{
+    fmt::{
+        format::{DefaultFields, FormatEvent, FormatFields, Writer},
+        FmtContext, FormattedFields, Layer,
+    },
+    layer::SubscriberExt,
+    registry::LookupSpan,
+    util::SubscriberInitExt,
+    Layer as LayerTrait, Registry,
+};
+
+pub type Result<T> = std::result::Result<T, TelemetryError>;
+
+/// Define a prelude to minimise a simple API
+///
+/// Exports the following names:
+///
+/// - `telemetry_init`
+/// - `TelemetryLayer`
+///
+/// - `trace!`
+/// - `debug!`
+/// - `info!`
+/// - `warn!`
+/// - `error!`
+///
+/// - `event!`
+/// - `span!`
+/// - `Level`
+pub mod prelude {
+    pub use crate::{
+        debug, error, event, info, span, telemetry_init, trace, warn, Level, TelemetryLayer,
+    };
+}
+
+/// Convenience function to do everything needed for telemetry in one step
+///
+/// This function does the following:
+/// - Creates a new `TelemetryLayer` `tracing` `Layer`
+/// - Sets it as the global default `tracing` `Subscriber`
+/// - Creates the root `span` called "main"
+/// - Enters the "main" `span`
+/// - Create a `FileWatcher` and starts it
+/// - Create a `SystemInfoWatcher` and starts it
+///
+/// Warning: this function should only be called in applications and not within
+/// libraries as it sets the global default subscriber, clobbering any set by
+/// the application.
+///
+/// ```rust
+/// use fuel_telemetry::prelude::*;
+///
+/// telemetry_init()?;
+/// ```
+pub fn telemetry_init() -> Result<()> {
+    static TELEMETRY_INIT: LazyLock<Result<(WorkerGuard, Span)>> = LazyLock::new(|| {
+        let guard = TelemetryLayer::new_global_default_with_watchers()?;
+        let main_span = span!(Level::ERROR, "main", telemetry = true);
```

Comment:
Why is this level ERROR?

Reply:
Good catch. Sorry, that's mostly muscle memory - I always turn production logging high so only errors log during production, but in this case telemetry will be lower than `ERROR`.

Reply:
No longer necessary as `telemetry_init()` has been removed.